whylog/assistant/pattern_match.py

```diff
@@ -1,21 +1,55 @@
-class ParamGroup(object):
-    def __init__(self, content, converter):
-        self.content = content
-        self.converter = converter
+import itertools
 
-    def __eq__(self, other):
-        return self.__dict__ == other.__dict__
+import six
+
+from whylog.assistant.validation_problems import InvalidConverterProblem, InvalidPrimaryKeyProblem
+from whylog.converters import get_converter
+from whylog.converters.exceptions import ConverterError
 
 
 class PatternMatch(object):
     def __init__(self, line_text, pattern, param_groups, primary_key=None):
-        # :type param_groups: dict[int, ParamGroup]
         self.line_text = line_text
         self.pattern = pattern
-        self.param_groups = param_groups
+        self.param_groups = param_groups  # dict[int, ParamGroup]
         self.primary_key = primary_key
         if primary_key is None:
             if not self.param_groups:
                 self.primary_key = []
             else:
                 self.primary_key = [min(self.param_groups.keys())]
+
+    def validate_primary_key(self):
+        group_numbers = self.param_groups.keys()
```

Comment:
I think the current code is ok, but be aware that You don't really return a list anyway: http://stackoverflow.com/questions/16819222/how-to-return-dictionary-keys-as-a-list-in-python-3-3